### PR TITLE
Remove window size/position restoration on help and about windows

### PR DESCRIPTION
**Pull request by ghost**
Removes the unecessary function to restore the about window size and
location for the about window when reopening the window.

Currently the window size and location is restored even after
application restart.

### CODE REVIEW

**Comment by jonasschnelli** [comment]
I didn't test, but the debug / rpc console has somehow the same code basis. I assume this PR will no longer store the position of the debug window as well?
Maybe you should take a look at `HelpMessageDialog::HelpMessageDialog` `about` boolean.

**Comment by jonasschnelli** [comment]
conceptional ACK on not storing the about window geometries.

**Comment by L-Cranston-Shadow** [comment]
@jonasschnelli I believe that does include the debug window as it currently is. Would there be any issue with splitting help message (i.e. the debug window) and about messages in to two classes? I see only [six references](https://github.com/bitcoin/bitcoin/search?l=cpp&q=HelpMessageDialog&utf8=%E2%9C%93) (not including translation files) to HelpMessageDialog.

**Comment by Diapolo** [comment]
Why? And also it's not sufficient to just remove the restoration code but also the save code...

File: src/qt/utilitydialog.cpp [hunk 1]
```diff
@@ -28,9 +28,6 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
     QDialog(parent),
     ui(new Ui::HelpMessageDialog)
 {
-    ui->setupUi(this);
```
Comment (jonasschnelli) [inline src/qt/utilitydialog.cpp]: Why did you remove this line? This will not work.

File: src/qt/utilitydialog.cpp [hunk 2]
```diff
@@ -28,9 +28,6 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
...
-    GUIUtil::restoreWindowGeometry("nHelpMessageDialogWindow", this->size(), this);
```
Comment (jonasschnelli) [inline src/qt/utilitydialog.cpp]: This needs a `if(!about)` and not a remove.

**Comment by jonasschnelli** [comment]
At the destructor `HelpMessageDialog::~HelpMessageDialog()` you must also make sure, geometry gets stored only if `about==false`. You can add a `fAbout` as private instance variable.

**Comment by Michagogo** [comment]
NAK on discarding debug window size/location. Indifferent regarding the about window (will this also affect the command line options window?), but the debug window is one which can actually be useful and which a user might reasonably want to have in a fixed location and not have to reposition it each time.

**Comment by jonasschnelli** [comment]
@Michagogo no no.
The Help- (command line) and Debug-window will keep storing the position.
This commit (https://github.com/jonasschnelli/bitcoin/commit/a16b79b16d9f13f13f13960e858d9416694a2ee4) would only affect the about window. No others.
I try to guide @L-Cranston-Shadow into this direction.

Not storing other windows geometry would be stupid. But the about window really don't need stored window geometries. The about window looks better without storing geometries and more like the "About Qt" window.

**Comment by Michagogo** [comment]
@jonasschnelli That commit sounds good, yeah. That's not what this PR is right now, though.

File: src/qt/utilitydialog.cpp (hunk 2 above)
Comment (laanwj) [inline src/qt/utilitydialog.cpp]: I'm not sure it is terribly useful for the help message dialog to store it's position either.

**Comment by laanwj** [comment]
Concept ACK - but please adapt your commit message to make it clear that only the HelpMessage/About dialog is affected and not every window. That kind of confused me and I see I'm not the only one.

File: src/qt/utilitydialog.cpp (hunk 2 above)
Comment (jonasschnelli) [inline src/qt/utilitydialog.cpp]: Indeed.

**Comment by L-Cranston-Shadow** [comment]
Sorry for the delay in replying, 

@Diapolo It was unclear in my original title but this PR is targeted specifically at the about window. The reasoning behind it was that the functionality of restoring the window size and position when reopening is unecessary so I, foolishly in retrospect, thought it would be a good way to dip my feet into both contributing to the project and getting back into C++.  

@jonasschnelli Removing that extra line was a mistake, I'll make sure to re-add it

@everyone I'll try to get a new commit up soon on this, I'm a little bit out of my depth both in terms of the specific code and the programming, some of which is stuff that I learned and I am having to pretty much relearn/remember or never learned and am having to teach myself on, so please bear with me.

**Comment by laanwj** [comment]
Oops. Something went wrong here, github reports a lot of added commits which were already merged.

Try to use rebase (e.g. git pull --rebase) instead of merge when you want to forward-port a pull request, so we can maintain a linear-ish history, as parallel merge commits make it hard to review.

**Comment by L-Cranston-Shadow** [comment]
See the above link, I somehow screwed something up and in the aftermath the system automatically closed this PR. Rather than email GitHub and ask for them to look into it, I figured it would be more expedient and almost as easy to create a new one.
